packages/blockchain-link-utils/src/solana.ts

```diff
@@ -85,6 +85,18 @@ export const getTxType = (
         return 'failed';
     }
 
+    const isTransfer = transaction.transaction.message.instructions.every(instruction => {
+        const isParsedInstruction = 'parsed' in instruction;
+        return isParsedInstruction && instruction.parsed.type === 'transfer';
+    });
+
+    // for now we support only transfers, so we interpret all other transactions as `unknown`
+    if (!isTransfer) {
+        return 'unknown';
+    }
```

Comment:
I realized that this implementation was quite unintuitive so I refactored it here [fix(blockchain-link): refactor solana getTxType](https://github.com/trezor/trezor-suite/pull/9498/commits/ed79cfdcd9f207533d155c1c19c0a5e82858e7a0)